src/vs/platform/storage/common/storage.ts

```diff
@@ -97,6 +97,17 @@ export interface IStorageService {
 	getNumber(key: string, scope: StorageScope, fallbackValue: number): number;
 	getNumber(key: string, scope: StorageScope, fallbackValue?: number): number | undefined;
 
+	/**
+	 * Retrieve an element stored with the given key from storage. Use
+	 * the provided `defaultValue` if the element is `null` or `undefined`.
+	 * The element will be converted to a `object` using `JSON.parse`.
+	 *
+	 * @param scope allows to define the scope of the storage operation
+	 * to either the current workspace only, all workspaces or all profiles.
+	 */
+	getObject(key: string, scope: StorageScope, fallbackValue: object): object;
```

Comment:
Would it make sense to introduce a `<T>` generic type here to automatically cast to the expected value?